packages/cli/scripts/util/bootstrap.js

```diff
@@ -82,12 +86,39 @@ async function processEvents(handler) {
   //    + callbackWaitsForEmptyEventLoop FALSE => callback value
 
   let result;
-  try {
-    result = await handler(EVENT, CONTEXT);
-  } catch (e) {
-    logger.debug("processEvents caught error");
-    return invokeErrorAndExit(e);
-  }
+  const {
+    groups: { root, parent },
+  } =
+    process.env._X_AMZN_TRACE_ID &&
+    XRAY_REGEX.exec(process.env._X_AMZN_TRACE_ID);
+  const segment = new AWSXRay.Segment(ORIG_HANDLER_PATH, root, parent);
+
+  await xrayNamespace.runAndReturn(async () => {
```

Comment:
Some notes on the MR, but essentially, this should be removed from here and live in its own function.

Segments should then be pushed into a buffer when returning from the handler, and periodically drained to prevent blocking awaits.